### ISSUE TITLE
questions about cosine distance in the code

### ISSUE THREAD

**Issue by icoz69** [wyharveychen/CloserLookFewShot#8]
hi, thank your for the nice work first.
i have some problems in the code that confuses me. To compute the cosine distance, the implementation is the following
`def forward(self, x):
        x_norm = torch.norm(x, p=2, dim =1).unsqueeze(1).expand_as(x)
        x_normalized = x.div(x_norm+ 0.00001)
        L_norm = torch.norm(self.L.weight.data, p=2, dim =1).unsqueeze(1).expand_as(self.L.weight.data)
        self.L.weight.data = self.L.weight.data.div(L_norm + 0.00001)
 
        cos_dist = self.L(x_normalized) #matrix product by forward function
        scores = self.scale_factor* (cos_dist) 

        return scores`

I notice that in the forward pass , the parameter of Linear function is replaced directly with the normed one. It is not simply a process to compute cosine distance.
It means at training time, the paramater theta is updated by: 
theta= normed (theta)
theta=theta - learning rate* theta
,not
theta=theta - learning rate* theta.

This is a bit different from the illustration in the paper.

**Comment by wyharveychen** [comment]
Yes, and for the complete issue, please see the discussion of issue#4.
I have marked the code more clearly, thanks!
